kotlin-ktor/src/test/java/com/baeldung/graphql/server/ConferenceTest.kt

```diff
@@ -0,0 +1,141 @@
+        val mockBody = serializeQuery(
+            FIND_BY_ID_QUERY
+        )
+        val call = client.post("graphql") {
+            header(HttpHeaders.ContentType, ContentType.Application.Json.toString())
+            setBody(mockBody)
+        }
+        with(call) {
+            assertEquals(HttpStatusCode.OK, status)
+            val response = readResponseAsMap(this)
+            assertNull((response["data"] as Map<*, *>)["conferenceById"])
+        }
+    }
+
+    @Test
+    fun `given one conference when fetching by id then return it`(): Unit = graphQlTestEnvironment { client ->
+        ConferenceRepository.save(Conference(id = null, name = "Conference", attendees = listOf()))
+        val mockBody = serializeQuery(
+            FIND_BY_ID_QUERY
+        )
+        val call = client.post("graphql") {
+            header(HttpHeaders.ContentType, ContentType.Application.Json.toString())
+            setBody(mockBody)
+        }
+        with(call) {
+            assertEquals(HttpStatusCode.OK, status)
+            val response = jacksonObjectMapper().readValue(bodyAsText(), Map::class.java)
+            assertNotNull((response["data"] as Map<*, *>)["conferenceById"])
+        }
+    }
+
+    @Test
+    fun `when creating a conference via mutations then should succeed`(): Unit = graphQlTestEnvironment { client ->
+        val mockBody = serializeQuery(
+            CREATE_CONFERENCE_QUERY
+        )
+        val call = client.post("graphql") {
+            header(HttpHeaders.ContentType, ContentType.Application.Json.toString())
+            setBody(mockBody)
+        }
+        with(call) {
+            assertEquals(HttpStatusCode.OK, status)
+            val response = jacksonObjectMapper().readValue(bodyAsText(), Map::class.java)
+            assertNotNull((response["data"] as Map<*, *>)["saveOrCreateConference"])
+            assertEquals("My Conference", ConferenceRepository.findById(0)?.name)
+        }
+    }
+
+    @Test
+    fun `when updating a conference via mutations then should succeed`(): Unit = graphQlTestEnvironment { client ->
+        ConferenceRepository.save(Conference(null, "Some Conference", listOf()))
+        val mockBody = serializeQuery(
+            UPDATE_CONFERENCE_QUERY
+        )
+        val call = client.post("graphql") {
+            header(HttpHeaders.ContentType, ContentType.Application.Json.toString())
+            setBody(mockBody)
+        }
+        with(call) {
+            assertEquals(HttpStatusCode.OK, status)
+            val response = jacksonObjectMapper().readValue(bodyAsText(), Map::class.java)
+            assertNotNull((response["data"] as Map<*, *>)["saveOrCreateConference"])
+            assertEquals("My Conference", ConferenceRepository.findById(0)?.name)
+        }
+    }
+
+    @Test
+    fun `when creating a conference then should trigger subscription`(): Unit = graphQlTestEnvironment { client ->
```

Comment:
Let's mention WebSocket in the test name to highlight it is doing something quite different than the other tests

Reply:
✅